Source/WebCore/css/parser/CSSPropertyParserConsumer+Timeline.cpp

```diff
@@ -150,5 +152,41 @@ RefPtr<CSSValue> consumeViewTimelineInset(CSSParserTokenRange& range, const CSSP
     });
 }
 
+static bool isAnimationRangeKeyword(CSSValueID id)
+{
+    return identMatches<CSSValueNormal, CSSValueCover, CSSValueContain, CSSValueEntry, CSSValueExit, CSSValueEntryCrossing, CSSValueExitCrossing>(id);
+}
+
+RefPtr<CSSValue> consumeAnimationRange(CSSParserTokenRange& range, const CSSParserContext& context, TimelineRangeStartOrEnd isStart)
+{
+    // https://drafts.csswg.org/scroll-animations-1/#propdef-animation-range-start
+    // normal | <length-percentage> | <timeline-range-name> <length-percentage>?
+    auto id = range.peek().id();
+
+    if (id == CSSValueNormal)
+        return consumeIdent(range);
```

Comment:
You may need to check that you're at the end of the range after consuming the ident, to ensure that something like "normal foo" or "normal 10px" is a parsing error. The tests should have some invalid value tests, which you should add to if they don't include all the invalid configurations.

Reply:
Right its not necessary to add a check. Did add some extra invalid tests though.